src/python/pygrackle/utilities/model_tests.py

```diff
@@ -135,22 +138,139 @@ def generate_model_sets(model_store):
 model_sets = {}
 generate_model_sets(model_sets)
 
-def get_model_set(model_name, parameter_index, input_index):
-    """
-    Create objects and variables to be used in testing one
-    of the Python example scripts.
+def _build_base_parser():
+    parser = argparse.ArgumentParser(
+        description=(
+            "This is a script that provides an illustrative example of how to "
+            "use Grackle can be used in a simplified model. When using the "
+            "script for learning, you should invoke it without any arguments. "
+            "Some subcommands are implemented to support using this script "
+            "for testing purposes"
+        )
+    )
+
+    subparsers = parser.add_subparsers(
+        required=True,
+        description = (
+            "these sub-commands support the use of the script's logic as part "
+            "of the test suite. We implement a command for evaluating the "
+            "logic with a parameter presets and an input-grid preset (both "
+            "sets of presets are associated with integers). Subcommands are "
+            "also provided to query the presets."
+        )
+    )
+
+    parser_run_test = subparsers.add_parser(
+        "run-test", description="execute the logic as part of a test-case"
+    )
+    parser_run_test.set_defaults(kind="run-test")
+    _kw = {'type' : int, 'required' : True}
+    parser_run_test.add_argument(
+        "--param-preset", help="index of the parameter preset", **_kw
+    )
+    parser_run_test.add_argument(
+        "--input-preset", help="index of the input-grid preset", **_kw
+    )
+    parser_run_test.add_argument(
+        "--skip-plot", action="store_true", help="skip any plotting"
+    )
+
+    _triples = [
+        ("show-param-preset", "show the parameter preset", True),
+        ("show-input-preset", "show the input-grid preset", True),
+        ("num-param-preset", "number of parameter presets", False),
+        ("num-input-preset", "number of input-grid presets", False),
+    ]
+    for name, descr, has_arg in _triples:
+        p = subparsers.add_parser(name, description=descr)
+        p.set_defaults(kind=name)
+        if has_arg:
+            p.add_argument("index", type=int, help="index of the preset")
```

Comment:
A potential simplification
```suggestion
    for name in ["param", "input"]:
        p = subparsers.add_parser(
            f"show-{name}-preset", description=f"show the {name} preset"
        )
        p.set_defaults(kind=f"show-{name}-preset")
        p.add_argument("index", type=int, help="index of the preset")
```